docs/main_differences.md

```diff
@@ -0,0 +1,21 @@
+# Main Differences Between DaCe and JaCe and JAX and JaCe
+
+Essentially JaCe is a frontend that allows DaCe to process JAX code, thus it has to be compatible with both, at least in some sense.
+We will now list the main differences between them, furthermore, you should also consult the ROADMAP.
+
+### JAX vs. JaCe:
+
+- JaCe always traces with enabled `x64` mode.
+  This is a restriction that might be lifted in the future.
+- JAX returns scalars as zero-dimensional arrays, JaCe returns them as array with shape `(1, )`.
```

Comment:
The main issue is that DaCe does not have a real concept of zero dimensional arrays, as far as I know.
Consider the following two functions
```python
@dace.program
def bar(a: dace.float64):
    return a + 1

@dace.program
def baz(a: dace.float64[1]):
    return a + 1
```
if you pass a zero dimensional array to `bar()` then it will be casted to an scalar, if you pass it to `baz()` an error will happen.
Furthermore, the binary interface of the SDFG can not return scalars, return values have to be arrays there is no way around that without patching the code generator and making a lot of changes to handle special cases.
So I decided to follow PEP20 and decided that this case is not special enough to change the rule.